caffe2/CMakeLists.txt

```diff
@@ -976,6 +977,41 @@ elseif(USE_CUDA)
     target_link_libraries(torch_cuda PRIVATE __caffe2_nccl)
     target_compile_definitions(torch_cuda PRIVATE USE_NCCL)
   endif()
+
+  # Use env var for these for now for prototyping purposes
+  set(USE_NVSHMEM $ENV{USE_NVSHMEM} CACHE BOOL "Enable NVSHMEM support")
+  set(NVSHMEM_HOME $ENV{NVSHMEM_HOME} CACHE PATH "Path to NVSHMEM build dir")
+
+  if(USE_NVSHMEM)
+    set(NVSHMEM_INCLUDE_DIR "${NVSHMEM_HOME}/include")
+    set(NVSHMEM_LIB_DIR "${NVSHMEM_HOME}/lib")
+
+    include_directories(${NVSHMEM_INCLUDE_DIR})
+
+    # Linking with nvshmem requires the source binary to be built with -rdc
+    # which is not viable for libtorch_cuda. So we isolate the linking of
+    # nvshmem in nvshmem_extension.
+    add_library(nvshmem_extension SHARED
+        "${TORCH_SRC_DIR}/csrc/distributed/c10d/nvshmem_extension.cu"
```

Comment:
i was gonna suggest having a dedicated folder for the nvshmem backend files so its clear to readers that they are built into a separate compilation unit.

however maybe its an annoyingly long path? `csrc/distributed/c10d/symm_mem/common` and `/nvshmem`

Reply:
Yeah, totally agree with a dedicated folder. The c10d folder is a mix of everything today, will refactor together with other libraries later.

Reply:
maybe we can have a symmetric memory dedicated folder?